rcl_executor/include/rcl_executor/let_executor.h

```diff
@@ -0,0 +1,289 @@
+ *   be set by calling {@link rcle_let_executor_set_timeout()} function (default value is 100ms)
+ *
+ * The static-LET executor performs the following actions:
+ * * initializes the wait_set with all handle of the array executor->handles
+ * * waits for new data from DDS queue with rcl_wait() with timeout executor->timeout_ns
+ * * takes all ready handles from the wait_set with rcl_take()
+ * * processes all handles in the order, how they were added to the executor with the respective add-functions
+ *   by calling respective callback (thus implementing first-read, process, semantic of LET)
+ *
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ *
+ * \param [inout] executor pointer to initialized executor
+ * \param[in] timeout_ns  timeout in millisonds
+ * \return `RCL_RET_OK` if spin_once operation was successful
+ * \return `RCL_RET_INVALID_ARGUMENT` if any parameter is a null pointer
+ * \return `RCL_RET_TIMEOUT` if rcl_wait() returned timeout (aka no data is avaiable during until the timeout)
+ * \return `RCL_RET_ERROR` if any other error occured
+ */
+rcl_ret_t
+rcle_let_executor_spin_some(
+  rcle_let_executor_t * executor,
+  const uint64_t timeout_ns);
+
+/**
+ *  The spin function checks for new data at DDS queue as long as ros context is available.
+ *  It calls {@link rcle_let_executor_spin_some()} as long as rcl_is_context_is_valid() returns true.
+ *
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ *
+ * \param [inout] executor pointer to initialized executor
+ * \return `RCL_RET_OK` if spin operation was successful
+ * \return `RCL_RET_INVALID_ARGUMENT` if executor is a null pointer
+ * \return `RCL_RET_ERROR` if any other error occured
+ */
+rcl_ret_t
+rcle_let_executor_spin(rcle_let_executor_t * executor);
+
+
+/**
+ *  The spin_period function checks for new data at DDS queue as long as ros context is available.
+ *  It is called every period nanoseconds.
+ *  It calls {@link rcle_let_executor_spin_some()} as long as rcl_is_context_is_valid() returns true.
+ *
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ *
+ * \param [inout] executor pointer to initialized executor
+ * \param [in] period in nanoseconds
+ * \return `RCL_RET_OK` if spin operation was successful
+ * \return `RCL_RET_INVALID_ARGUMENT` if executor is a null pointer
+ * \return `RCL_RET_ERROR` if any other error occured
+ */
+rcl_ret_t
+rcle_let_executor_spin_period(rcle_let_executor_t * executor, const uint64_t period);
+
+/**
+ * macro to print errors
+ */
+
+#define PRINT_RCL_ERROR(rclc, rcl) \
```

Comment:
if i were, i would change the definition name into PRINT_RCL_LET_ERROR, looks more related to LET and less rcl generic.

Reply:
resolved in commit 861d16a164e8cd51760576a77ad8cd6e1449bb07